src/Kiota.Builder/Refiners/ShellRefiner.cs

```diff
@@ -103,6 +105,20 @@ private static void CreateCommandBuilders(CodeElement currentElement)
         CrawlTree(currentElement, CreateCommandBuilders);
     }
 
+    private static void RemoveUnusedParameters(CodeClass currentClass)
+    {
+        var unusedPropKinds = new[] { CodePropertyKind.RequestAdapter };
+        var unusedParamKinds = new[] { CodeParameterKind.RequestAdapter };
+        var requestAdapters = currentClass.Properties.Where(p => p.IsOfKind(unusedPropKinds));
```

Comment:
once the kinds are moved as constants, you'll be able to switch to a static lambda